### PR TITLE
2048 aytq migrate production infra

### PR DESCRIPTION
**Pull request by shaheislamdfe**
### Context

Adding additional infra necessary for production environment, including HA for Postgres, added replicas for web app deployment and worker application deployment.

### Changes proposed in this pull request

Adding additional infra necessary for production environment, including HA for Postgres, added replicas for web app deployment and worker application deployment.

### Guidance to review

Dev team to validate production infra and test worker functionality.

### Link to Trello card

https://trello.com/c/S5we2pWZ/2027-aytq-production-environment

### Checklist

- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by shaheislamdfe** [comment]
[AYTQ: production environment](https://trello.com/c/S5we2pWZ/2027-aytq-production-environment)

File: terraform/application/application.tf [hunk 1]
```diff
@@ -40,9 +40,30 @@ module "web_application" {
...
+  environment  = var.environment
+  service_name = var.service_name
+
+  cluster_configuration_map  = module.cluster_data.configuration_map
+  kubernetes_config_map_name = module.application_configuration.kubernetes_config_map_name
+  kubernetes_secret_name     = module.application_configuration.kubernetes_secret_name
+  replicas                   = var.worker_replicas
+  docker_image               = var.docker_image
+  enable_logit               = true
+  enable_gcp_wif             = true
```
Comment (saliceti) [inline terraform/application/application.tf]: GCP WIF is not enabled yet, let's not add it here

File: terraform/application/application.tf [hunk 2]
```diff
@@ -40,9 +40,30 @@ module "web_application" {
...
+  namespace    = var.namespace
+  environment  = var.environment
+  service_name = var.service_name
+
+  cluster_configuration_map  = module.cluster_data.configuration_map
+  kubernetes_config_map_name = module.application_configuration.kubernetes_config_map_name
+  kubernetes_secret_name     = module.application_configuration.kubernetes_secret_name
+  replicas                   = var.worker_replicas
+  docker_image               = var.docker_image
+  enable_logit               = true
```
Comment (saliceti) [inline terraform/application/application.tf:68]: Could we enable Logit for the webapp as well?

File: terraform/application/database.tf [hunk 3]
```diff
@@ -11,6 +11,8 @@ module "postgres" {
   use_azure                   = var.deploy_azure_backing_services
   azure_enable_monitoring     = var.enable_monitoring
   azure_enable_backup_storage = var.enable_postgres_backup_storage
+  azure_sku_name              = var.postgres_flexible_server_sku
+  azure_enable_high_availability = var.postgres_enable_high_availability
```
Comment (saliceti) [inline terraform/application/database.tf:15]: Please fix indentation

File: terraform/application/variables.tf [hunk 4]
```diff
@@ -66,7 +66,25 @@ variable "evidence_container_retention_in_days" {
...
+  default     = 1
+}
+variable "worker_replicas" {
+  description = "number of replicas of the workers"
+  default     = 1
+}
+variable "postgres_enable_high_availability" {
+  default = false
+}
+variable "azure_enable_high_availability" {
```
Comment (saliceti) [inline terraform/application/variables.tf]: Is this required?

**Review by saliceti — APPROVED** [review]

**Pull request merged**